pkgs/dart_mcp/lib/src/server/logging_support.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'server.dart';
+
+/// A mixin for MCP servers which support the `logging` capability.
+///
+/// See https://spec.modelcontextprotocol.io/specification/2025-03-26/server/utilities/logging/.
+base mixin LoggingSupport on MCPServer {
+  /// The current logging level, defaults to [LoggingLevel.warning].
+  LoggingLevel loggingLevel = LoggingLevel.warning;
+
+  @override
+  FutureOr<InitializeResult> initialize(InitializeRequest request) async {
+    _peer.registerMethod(
+      SetLevelRequest.methodName,
+      convertParameters(handleSetLevel),
+    );
+
+    final result = await super.initialize(request);
+    result.capabilities.logging ??= Logging();
+    return result;
```

Comment:
nit: this can be simplified, although maybe less readable, so I will leave up to you.
```dart
return (await super.initialize(request))
  ..capabilities.logging ?? = Logging();
```